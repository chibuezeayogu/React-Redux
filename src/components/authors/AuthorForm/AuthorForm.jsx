import React, { PropTypes } from 'react';
import {
	object, array, func, bool, shape
} from 'prop-types';
import TextInput from '../../common/TextInput/TextInput';

const AuthorForm = ({
	author, onSave, onChange, loading, errors 
}) => (
	<form>
		<TextInput
			name="firstName"
			label="First Name"
			value={author.firstName}
			onChange={onChange}
			error={errors.firstName}/>

		<TextInput
			name="lastName"
			label="Last Name"
			value={author.lastName}
			onChange={onChange}
			error={errors.lastName}/>

		<button
			disabled={loading}
			className="btn btn-primary"
			onClick={onSave}
		>
			{loading ? 'Saving...' : 'Save'}
		</button>
	</form>
);

AuthorForm.propTypes = {
	author: shape({}).isRequired,
	onSave: func.isRequired,
	onChange: func.isRequired,
	loading: bool,
	errors: object
};

export default AuthorForm;