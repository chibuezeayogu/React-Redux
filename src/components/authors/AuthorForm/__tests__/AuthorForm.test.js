import React from 'react';
import { shallow } from 'enzyme';

import AuthorForm from '../AuthorForm';

describe('AboutPage Component', () => {
	const props = {
		onSave: jest.fn(),
		onChange: jest.fn(),
		loading: false,
		author: {
			id: '',
			firstName: '',
			lastName: ''
		},
		errors: {
			firstName: '',
			lastName: ''
		}
	};

	const wrapper = shallow(<AuthorForm {...props} />);

	it('should render correctly', () => {
		expect(wrapper.exists()).toBeTruthy();
		expect(wrapper).toMatchSnapshot();
	});

	it('should render form, TextInput, button', () => {
		expect(wrapper.find('form')).toHaveLength(1);
		expect(wrapper.find('button')).toHaveLength(1);
		expect(wrapper.find('button').text()).toEqual('Save');
		expect(wrapper.find('TextInput')).toHaveLength(2);
		expect(wrapper.find('TextInput')).toHaveLength(2);
	});
});