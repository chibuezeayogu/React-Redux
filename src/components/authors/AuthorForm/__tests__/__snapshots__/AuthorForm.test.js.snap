// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutPage Component should render correctly 1`] = `
<form>
  <TextInput
    error=""
    label="First Name"
    name="firstName"
    onChange={[MockFunction]}
    value=""
  />
  <TextInput
    error=""
    label="Last Name"
    name="lastName"
    onChange={[MockFunction]}
    value=""
  />
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[MockFunction]}
  >
    Save
  </button>
</form>
`;
